har_dataset.py

```diff
@@ -685,34 +715,52 @@ def __prepare_motionsense__(path, split):
         [pl.col(c) + 2 * std_values[c]
             for c in set(mean_values.columns).intersection(std_values.columns)]
     )
-
+    
     print(f'Parsing {len(all_files)} files...')
     recordings = []
+    
     for i, file in enumerate(all_files):
-        subject = int(file.stem.split('_')[1])
-        class_name_full = str(file.parent.name)
-        class_name = class_name_full.split('_')[0]
+        # skip subjects according to split id list
+        if half_data==True:
+            if subject not in ids[split]:
+                continue
+            subject = int(file.stem.split('_')[1])
+            class_name_full = str(file.parent.name)
+            class_name = class_name_full.split('_')[0]
```

Comment:
These three, are the same as in else below. Add them to the beginning of the for loop, then subject also exists